tstrait/simulate_effect_size.py

```diff
@@ -233,21 +227,44 @@ def sim_trait(ts, model, *, num_causal=None, alpha=None, random_seed=None):
     """
     ts = _check_instance(ts, "ts", tskit.TreeSequence)
     model = _check_instance(model, "model", tstrait.TraitModel)
-    num_causal = 1 if num_causal is None else num_causal
-    num_causal = _check_int(num_causal, "num_causal", minimum=1)
+    if num_causal is not None and causal_sites is not None:
+        raise ValueError("Cannot specify both num_causal and causal_sites")
+    if num_causal is None and causal_sites is None:
+        num_causal = 1
+
+    if isinstance(causal_sites, numbers.Real):
```

Comment:
I don't think we want to do this - you're allowing for a single scalar here as causal_sites, right? If the user wants a single site then it's trivial to do ``causal_sites=[10]`` e.g., and it would be very easy to make the mistake of doing

``sim_trait(ts, causal_sites=10)`` instead of ``sim_trait(ts, num_causal=10)``

Reply:
It is fixed now.